array.h

```diff
@@ -908,6 +917,43 @@ template <size_t Rank, size_t... Is>
 using enable_if_permutation = std::enable_if_t<sizeof...(Is) == Rank && all(Is < Rank...) &&
                                                product((Is + 2)...) == factorial(Rank + 1)>;
 
+template <index_t DimIndex, index_t Min, index_t Extent, index_t Stride,
```

Comment:
If you make the template parameter `class DimDst`, I think you can get the parameters via `DimDst::Min`, `DimDst::Extent`, `DimDst::Stride`. Then you don't need to make an instance of DimDst. It probably wouldn't affect the generated code (much) but I think it's better to avoid constructing dims unless needed.

Reply:
Done.

Reply:
Done. Thank you for all the advice!